lib/tpm2_hmac_auth.c

```diff
@@ -125,35 +125,37 @@ bool tpm2_get_entity_name(TSS2_SYS_CONTEXT *sapi_context, TPM2_HANDLE entity_han
     bool res;
     uint8_t temp[4];
     //Get Entity names from handles
-    switch(entity_handle & 0xFF000000) {
-        case 0x01000000: res = tpm2_util_nv_read_public(sapi_context, 
-                                                        entity_handle, 
-                                                        &nv_public,
-                                                        entity_name);
-                    if (!res) {
-                        LOG_ERR("Failed reading NV public when calculating name");
-                        return false;
-                    }
-                    break;
-        case 0x80000000:
-        case 0x81000000: res = tpm2_read_transient_persistent_obj_name(sapi_context,
-                                    entity_handle, entity_name);
-                         if (!res) {
-                            LOG_ERR("Failed reading Object public when calculating name");
-                            return false;
-                        }
-                        break;
-        case 0x00000000:
-        case 0x02000000:
-        case 0x03000000:
-        case 0x40000000: entity_name->size = sizeof(TPM2_HANDLE);
-                         memcpy(temp, (uint8_t *)&entity_handle, sizeof(TPM2_HANDLE));
-                         for(unsigned i=0; i<sizeof(TPM2_HANDLE); i++) {
-                            entity_name->name[i] = temp[3-i];
-                         }
-                         break;
-        default:         LOG_ERR("Handle not valid for HMAC auth");
-                         return false;
+    switch((entity_handle & TPM2_HR_RANGE_MASK)>>TPM2_HR_SHIFT) {
```

Comment:
No need for that bit shifting I believe, you could just do:

```
switch (entity_handle & TPM2_HR_RANGE_MASK) {
        case TPM2_PT_TPM2_HR_NV_INDEX:
        ....
}
```

Reply:
@idesai and also didn't do the suggested change.

Reply:
Fixed